fix `nix build` for the X11 version

### Description

Since [this run](https://github.com/espanso/espanso/actions/runs/17812567948), the `nix build` step for X11 is failing with the following log:

### Logs

```
building '/nix/store/adix02z0n07sj2qszfw5cv48210vxy5j-cargo-vendor-dir.drv'...
copying path '/nix/store/hcginxlb1vivl6v6vjmi61s96082270d-rustc-wrapper-1.89.0' from 'https://cache.nixos.org/'...
copying path '/nix/store/xq6ns6hdgl7fdshisp5mb65f1jlysm0h-cargo-1.89.0' from 'https://cache.nixos.org/'...
copying path '/nix/store/njcdzjkw51b1g2w60s9vrd24m7lb8a94-auditable-cargo-1.89.0' from 'https://cache.nixos.org/'...
building '/nix/store/9kk1ahr0v4s708mp27gz2qa8wvcsr54b-espanso-2.2.5.drv'...
error: Cannot build '/nix/store/9kk1ahr0v4s708mp27gz2qa8wvcsr54b-espanso-2.2.5.drv'.
       Reason: builder failed with exit code 101.
       Output paths:
         /nix/store/qx72g0v1gn7gxf8397p9q2f1lix0xyv3-espanso-2.2.5
       Last 25 log lines:
       >   cargo:warning=src/x11/native/clip/clip_x11.cpp:10:10: fatal error: xcb/xcb.h: No such file or directory
       >   cargo:warning=   10 | #include <xcb/xcb.h>
       >   cargo:warning=      |          ^~~~~~~~~~~
       >   cargo:warning=compilation terminated.
       >
       >   --- stderr
       >
       >
       >   error occurred in cc-rs: command did not execute successfully (status code exit status: 1): LC_ALL="C" "/nix/store/95k9rsn1zsw1yvir8mj824ldhf90i4qw-gcc-wrapper-14.3.0/bin/c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "src/x11/native/clip" "-I" "src/x11/native" "-Wall" "-Wextra" "-fno-omit-frame-pointer" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/espanso-clipboard-a169b9e0c9d44d48/out/824a84b596d5b425-clip_x11.o" "-c" "src/x11/native/clip/clip_x11.cpp"
       >
       >
       > warning: build failed, waiting for other jobs to finish...
       > warning: espanso-inject@0.1.0: src/evdev/native.c: In function 'emit':
       > warning: espanso-inject@0.1.0: src/evdev/native.c:54:5: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [-Wunused-result]
       > warning: espanso-inject@0.1.0:    54 |     write(fd, &ie, sizeof(ie));
       > warning: espanso-inject@0.1.0:       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > warning: espanso-inject@0.1.0: cc1: warning: src/x11/xdotool/vendor/xdo.h: not a directory
       > warning: espanso-inject@0.1.0: cc1: warning: src/x11/xdotool/vendor/xdo_util.h: not a directory
       > warning: espanso-inject@0.1.0: src/x11/xdotool/vendor/xdo.c: In function 'xdo_get_active_modifiers':
       > warning: espanso-inject@0.1.0: src/x11/xdotool/vendor/xdo.c:1773:27: warning: pointer 'keys' used after 'realloc' [-Wuse-after-free]
       > warning: espanso-inject@0.1.0:  1773 |                     *keys = realloc(keys, keys_size * sizeof(charcodemap_t));
       > warning: espanso-inject@0.1.0:       |                     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > warning: espanso-inject@0.1.0: src/x11/xdotool/vendor/xdo.c:1773:29: note: call to 'realloc' here
       > warning: espanso-inject@0.1.0:  1773 |                     *keys = realloc(keys, keys_size * sizeof(charcodemap_t));
       > warning: espanso-inject@0.1.0:       |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       For full logs, run:
         nix log /nix/store/9kk1ahr0v4s708mp27gz2qa8wvcsr54b-espanso-2.2.5.drv
```

### Self-service

- [ ] I'd be willing to fix this bug myself.

--- a/espanso-config/src/config/resolve.rs
+++ b/espanso-config/src/config/resolve.rs
@@ -473,13 +473,13 @@ impl ResolvedConfig {
 
         if let Some(yaml_includes) = config.includes.as_ref() {
             for include in yaml_includes {
-                includes.insert(include.to_string());
+                includes.insert(include.clone());
             }
         }
 
         if let Some(extra_includes) = config.extra_includes.as_ref() {
             for include in extra_includes {
-                includes.insert(include.to_string());
+                includes.insert(include.clone());
             }
         }
 
@@ -491,13 +491,13 @@ impl ResolvedConfig {
 
         if let Some(yaml_excludes) = config.excludes.as_ref() {
             for exclude in yaml_excludes {
-                excludes.insert(exclude.to_string());
+                excludes.insert(exclude.clone());
             }
         }
 
         if let Some(extra_excludes) = config.extra_excludes.as_ref() {
             for exclude in extra_excludes {
-                excludes.insert(exclude.to_string());
+                excludes.insert(exclude.clone());
             }
         }
 

--- a/espanso-detect/src/evdev/device.rs
+++ b/espanso-detect/src/evdev/device.rs
@@ -94,7 +94,7 @@ impl Device {
     }
 
     pub fn get_path(&self) -> String {
-        self.path.to_string()
+        self.path.clone()
     }
 
     pub fn read(&self) -> Result<Vec<RawInputEvent>> {

--- a/espanso-engine/src/process/middleware/match_exec.rs
+++ b/espanso-engine/src/process/middleware/match_exec.rs
@@ -55,7 +55,7 @@ impl Middleware for MatchExecRequestMiddleware<'_> {
             // Inject the request args into the detected matches
             for m in &mut matches {
                 for (key, value) in &m_event.args {
-                    m.args.insert(key.to_string(), value.to_string());
+                    m.args.insert(key.clone(), value.clone());
                 }
             }
 

--- a/espanso-package/src/archive/mod.rs
+++ b/espanso-package/src/archive/mod.rs
@@ -116,7 +116,7 @@ impl From<&ArchivedPackage> for PackageSpecifier {
     fn from(package: &ArchivedPackage) -> Self {
         match &package.source {
             PackageSource::Hub => Self {
-                name: package.manifest.name.to_string(),
+                name: package.manifest.name.clone(),
                 ..Default::default()
             },
             PackageSource::Git {
@@ -124,8 +124,8 @@ impl From<&ArchivedPackage> for PackageSpecifier {
                 repo_branch,
                 use_native_git,
             } => Self {
-                name: package.manifest.name.to_string(),
-                git_repo_url: Some(repo_url.to_string()),
+                name: package.manifest.name.clone(),
+                git_repo_url: Some(repo_url.clone()),
                 git_branch: repo_branch.clone(),
                 use_native_git: *use_native_git,
                 ..Default::default()

--- a/espanso-render/src/extension/script.rs
+++ b/espanso-render/src/extension/script.rs
@@ -154,7 +154,7 @@ impl Extension for ScriptExtension {
                     ExtensionResult::Success(ExtensionOutput::Single(output))
                 }
                 Err(error) => ExtensionResult::Error(
-                    ScriptExtensionError::ExecutionFailed(args[0].to_string(), error.into()).into(),
+                    ScriptExtensionError::ExecutionFailed(args[0].clone(), error.into()).into(),
                 ),
             }
         } else {

--- a/espanso-render/src/extension/shell.rs
+++ b/espanso-render/src/extension/shell.rs
@@ -220,7 +220,7 @@ impl Extension for ShellExtension {
                     shell
                 } else {
                     return ExtensionResult::Error(
-                        ShellExtensionError::InvalidShell(shell_param.to_string()).into(),
+                        ShellExtensionError::InvalidShell(shell_param.clone()).into(),
                     );
                 }
             } else {
@@ -286,7 +286,7 @@ impl Extension for ShellExtension {
                     ExtensionResult::Success(ExtensionOutput::Single(output))
                 }
                 Err(error) => ExtensionResult::Error(
-                    ShellExtensionError::ExecutionFailed(cmd.to_string(), error.into()).into(),
+                    ShellExtensionError::ExecutionFailed(cmd.clone(), error.into()).into(),
                 ),
             }
         } else {

--- a/espanso-render/src/renderer/mod.rs
+++ b/espanso-render/src/renderer/mod.rs
@@ -280,13 +280,13 @@ mod tests {
                 (params.get("name"), params.get("value"))
             {
                 let mut map = HashMap::new();
-                map.insert(name.to_string(), value.to_string());
+                map.insert(name.clone(), value.clone());
                 return ExtensionResult::Success(ExtensionOutput::Multiple(map));
             }
             // If the "read" param is present, echo the value of the corresponding result in the scope
             if let Some(Value::String(string)) = params.get("read") {
                 if let Some(ExtensionOutput::Single(value)) = scope.get(string.as_str()) {
-                    return ExtensionResult::Success(ExtensionOutput::Single(value.to_string()));
+                    return ExtensionResult::Success(ExtensionOutput::Single(value.clone()));
                 }
             }
             if params.get("abort").is_some() {

--- a/nix/espanso.nix
+++ b/nix/espanso.nix
@@ -28,11 +28,9 @@ let
 in
 rustPlatform.buildRustPackage {
   pname = espansoCargo.package.name;
-  version = espansoCargo.package.version;
+  inherit (espansoCargo.package) version;
 
-  cargoLock = {
-    lockFile = ../Cargo.lock;
-  };
+  cargoLock.lockFile = ../Cargo.lock;
   src = lib.fileset.toSource {
     root = ../.;
     fileset = lib.fileset.unions [
@@ -43,7 +41,6 @@ rustPlatform.buildRustPackage {
       ../Cargo.toml
     ];
   };
-  doCheck = true;
 
   buildInputs = [
     libpng

--- a/nix/espanso.nix
+++ b/nix/espanso.nix
@@ -16,6 +16,7 @@
   wxGTK32,
   xclip,
   xdotool,
+  xorg,
   waylandSupport ? false,
   buildType ? "release",
 }:
@@ -56,6 +57,7 @@ rustPlatform.buildRustPackage {
     wl-clipboard
   ]
   ++ lib.optionals x11Support [
+    xorg.libxcb.dev
     libX11
     libXi
     libXtst

